plugin/PDFContextMenuHighlight.qml

```diff
@@ -0,0 +1,137 @@
+        font.pixelSize: Theme.fontSizeSmall
+        wrapMode: Text.Wrap
+        elide: Text.ElideRight
+        maximumLineCount: 2
+        color: Theme.highlightColor
+        opacity: .6
+        text: {
+            if (contextMenuHighlight.annotation !== null
+                && contextMenuHighlight.annotation !== undefined
+                && contextMenuHighlight.annotation.contents != "") {
+                return (contextMenuHighlight.annotation.author != ""
+                        ? "(" + contextMenuHighlight.annotation.author + ") " : "")
+                + contextMenuHighlight.annotation.contents
+            } else {
+                return ""
+            }
+        }
+    }
+    Row {
+        height: Theme.itemSizeExtraSmall
+        Repeater {
+            id: colors
+            model: ["#db431c", "#ffff00", "#8afa72", "#00ffff",
+                    "#3828f9", "#a328c7", "#ffffff", "#989898",
+                    "#000000"]
+            delegate: Rectangle {
+                width: contextMenuHighlight.width / colors.model.length
+                height: parent.height
+                color: modelData
+                MouseArea {
+                    anchors.fill: parent
+                    onClicked: {
+                        contextMenuHighlight.hide()
+                        contextMenuHighlight.annotation.color = color
+                        highlightColorConfig.value = modelData
+                    }
+                }
+            }
+        }
+    }
+    Row {
+        height: Theme.itemSizeExtraSmall
+        Repeater {
+            id: styles
+            model: [{"style": PDF.HighlightAnnotation.Highlight,
+                     "label": "abc"},
+                    {"style": PDF.HighlightAnnotation.Squiggly,
+                     "label": "a̰b̰c̰"},
+                    {"style": PDF.HighlightAnnotation.Underline,
+                     "label": "<span style=\"text-decoration:underline\">abc</span>"},
+                    {"style": PDF.HighlightAnnotation.StrikeOut,
+                     "label": "<span style=\"text-decoration:line-through\">abc</span>"}]
+            delegate: BackgroundItem {
+                id: bgStyle
+                width: contextMenuHighlight.width / styles.model.length
+                height: parent.height
+                onClicked: {
+                    contextMenuHighlight.hide()
+                    contextMenuHighlight.annotation.style = modelData["style"]
+                    highlightStyleConfig.value = highlightStyleConfig.fromEnum(modelData["style"])
+                }
+                Label {
+                    anchors.centerIn: parent
+                    text: modelData["label"]
+                    textFormat: Text.RichText
+                    color: bgStyle.highlighted
+                           || (contextMenuHighlight.annotation !== undefined
+                               && contextMenuHighlight.aannotation !== null
```

Comment:
Now that you mentioned these, property variant is deprecated and should be changed to property var: https://wiki.qt.io/Property-var - besides annotations, there's property variant _menu on ContextMenuHook.

For import, don't think there's anything requiring "as PDF", so just drop it if you want.